### PR TITLE
Enumerable methods

### PR DESCRIPTION
**Pull request by Cathella**
- This project implements enumerable methods in a different way than the usual but having the same output in Ruby. The project requires me to reproduce the following methods in other ways since in ruby a code implementation can be written in very many other ways. #my_each, #my_each_with_index, #my_select, #my_all?, #my_any?, #my_none?, #my_count, #my_map, #my_inject.

### CODE REVIEW

**Review by mosesogwo — CHANGES_REQUESTED** [review]
**Hi @Cathella, Great work:+1:. Please consider the following:**
- For **my_all?, my_any? and my_none?**, Your methods do not cover cases when **a class**, **a regex** or any pattern that is not a class or regex is passed as argument. For instance, `[1, 2, 3].my_all?(Integer)` should return `true` but your method currently throws an error.

- **my_inject** should work for a range of values
Say, ` (0..3).my_inject {|acc, x| acc + x }` should return `6` and not throw an error.

- **my_inject** should also be able to take a single argument -  a symbol - and combine the elements of the enum with the binary operation specified by that symbol.
`[1, 2, 3].inject(:+)` should return `6` and not throw an error.

- Use 2 spaces for your indentation instead of 4.


_**Kindly make these changes and submit another code review request.**_

**Any questions? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes Requested.

File: Enumerable.rb [hunk 1]
```diff
@@ -1,114 +1,131 @@
...
+  # rubocop:disable Metrics/PerceivedComplexity
+  def my_all?(arg = nil)
+    result = true
+    my_each do |i|
+      if result
+        if arg.is_a?(Class) || arg.is_a?(Regexp)
+          result = i.is_a?(arg) unless arg.is_a?(Regexp)
+          result = !i.match(arg).nil? if arg.is_a?(Regexp)
+        elsif arg
+          result = i == arg1
```
Comment (uryelah) [inline Enumerable.rb]: - [ ] There's a typo in the line 38, inside my_all?
It should be just arg and not arg1 right?

File: Enumerable.rb [hunk 2]
```diff
@@ -1,114 +1,131 @@
...
-                result = operation.call(result, i)
-            end
-        end 
-        result
+  # rubocop:disable Metrics/CyclomaticComplexity
+  # rubocop:disable Metrics/PerceivedComplexity
+  def my_inject(arg = nil, arg2 = nil)
+    result = is_a?(Range) ? min : self[0]
+    if arg2.is_a?(Symbol) || arg2.is_a?(String)
+      my_each_with_index { |element, i| result = result.send(arg2, element) if i.positive? }
```
Comment (uryelah) [inline Enumerable.rb]: - [ ] Your my_inject method is still not working with ranges. 
To understand why test your code with a range like (5..10) and see what happens.
Take also a moment to test in the irb console what are the properties that arrays have, and are being used by your code, and ranges don't have? How can you make the methods work with ranges then?

**Review by uryelah — CHANGES_REQUESTED** [review]
Hello **Nakitto Catherine**, I'm Sarah, your reviewer for this PR 👩‍🚀  

**First of all good job tidying up your code and fixing some of the previous bugs🌟**

There are still some tests your methods are not passing, but you got this!💪 


## Required Changes 🏗 👷‍♀ 

Please read the comments below with atention and fix the pointed issues:
---------

**Comment by uryelah** [comment]
**Kindly fix the REQUIRED issues and send another code review request.**

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

File: Enumerable.rb (hunk 2 above)
Comment (uryelah) [reply Enumerable.rb]: This article is a bit lengthy but it has some interesting comparisons between arrays and ranges and might help your solution and understanding of ruby: [https://www.codesdope.com/ruby-array/](https://www.codesdope.com/ruby-array/)

**Review by pelzolga123 — APPROVED** [review]
Approved.

**Comment by pelzolga123** [comment]
## Great work, Congratulation :+1: Project Approved.
You did all changes required previously. Go ahead and merge to development.
_Goodluck in your next project_ *Happy coding*

**Pull request merged**